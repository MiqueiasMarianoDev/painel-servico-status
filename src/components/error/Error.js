import React from 'react'

export const Error = ({mensagem}) => {
    return (
      <div className="erro">
        <p>Oops, ocorreu um erro:</p>
        <p>{mensagem}</p>
      </div>
    );
  
}