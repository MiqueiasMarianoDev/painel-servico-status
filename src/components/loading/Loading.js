import React from 'react'
import { Animation } from '../icones/Animation';
import LoadingAnimationData from '../animation/animation_llbrzglt.json'

export const Loading = () => {
  return (
    <div className="loading">
      <div className="spinner"></div>
      <p>Carregando...</p> 
      <Animation animationData={LoadingAnimationData} width={50} height={50}loop={true}/>
    </div>
  );
}