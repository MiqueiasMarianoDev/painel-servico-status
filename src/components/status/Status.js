import React from 'react'
import { OkIcon } from '../icones/OkIcon';
import { WarningIcon } from '../icones/WarningIcon';
import { ErrorIcon } from '../icones/ErrorIcon';

export const Status = ({status}) => {

  switch (status.props.cell.value) {
    case 'up':
      return <OkIcon />
    case 'warning': 
      return <WarningIcon />
    case 'error':
      return <ErrorIcon />
    default:
      return <p>null</p>
  }
}