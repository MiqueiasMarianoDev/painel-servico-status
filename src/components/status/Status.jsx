import React from 'react'
import './Status.module.css'
import { Icones } from '../icones/Icones';

export const Status = ({_status}) => {

  const status = _status.props.cell.value;

  let color = '';

  switch(status) {
    case 'up':
      color = 'green';
      break;
    case 'warning':
      color = 'gray';
      break;
    case 'error':
      color = 'red';
      break;
    default: 
      color = '';
  }

  return (
    <div className={`status ${color}`}>
      <Icones status={status} />
    </div>
  );

  }


