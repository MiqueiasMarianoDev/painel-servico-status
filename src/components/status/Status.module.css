.hjistatus {
    display: inline-block;
    padding: 4px 8px;
    border-radius: 4px;
    font-size: 14px;
  }
  
.green {
    color: green;
  }
.red {
    background: grey;
    color: #fff;
  }
.red {
    background: #f44336;
    color: #fff;
  }