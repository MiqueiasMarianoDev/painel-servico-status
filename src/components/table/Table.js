import React, { useMemo } from 'react'
import { useTable } from 'react-table/dist/react-table.development'
import './Table.module.css'
import { Status } from '../status/Status';
import { Loading } from '../loading/Loading';
import { Error } from '../error/Error';


export const Table = ({_columns, _data, loading, error}) => {
  const columns = useMemo(() => _columns, [_columns]);
  const data = useMemo(() => _data, [_data]);

  const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    rows,
    prepareRow
  } = useTable({
    columns,
    data
  })

  return (
    <>
        <table {...getTableProps()}>
            <thead>
            {headerGroups.map(headerGroup => (
                <tr {...headerGroup.getHeaderGroupProps()}>
                {headerGroup.headers.map(column => (
                    <th {...column.getHeaderProps()}>{column.render('Header')}</th>
                ))}
                </tr>
            ))}
            </thead>
            <tbody {...getTableBodyProps()}>
                {rows && rows.map(row => {
                    prepareRow(row)
                    return (
                        <tr {...row.getRowProps()}>
                            {row.cells.map(cell => {
                                if (cell.column.id === "status") {
                                  return <td {...cell.getCellProps()}><Status _status={cell.render('Cell')} /></td>
                                }
                                return <td {...cell.getCellProps()}>{cell.render('Cell')}</td>
                            })}
                        </tr>
                    )
                })
                }
            </tbody>   
        </table>
        {loading && (
          <Loading />
        )}
        {error &&
          <Error />
        }


    </>
  )
}
Table.defaultProps = {
  loading: false,
  error: false
}