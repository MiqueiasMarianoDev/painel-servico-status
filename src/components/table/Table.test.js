import React from 'react';
import { render, screen } from '@testing-library/react';
import { Table } from './Table';
import { Status } from '../status/Status';
import { Loading } from '../loading/Loading';
import { Error } from '../error/Error';

const columns = [{
  Header: 'Coluna 1',
  accessor: 'col1' 
}]

const data = [
  { col1: 'Valor 1' }, 
  { col1: 'Valor 2' }
]

describe('Table', () => {

  test('renders table element', () => {
    render(<Table _columns={columns} _data={data} />);
    const table = screen.getByRole('table');
    expect(table).toBeInTheDocument();
  });

  test('renders header', () => {
    render(<Table _columns={columns} _data={data} />);
    const header = screen.getByRole('columnheader', { name: /Coluna 1/i});
    expect(header).toBeInTheDocument();
  });

  test('renders rows', () => {
    render(<Table _columns={columns} _data={data} />);
    const rows = screen.getAllByRole('row');
    expect(rows.length).toBe(3); // inclui header
  });
/*
  test('renders data in rows', () => {
    render(<Table _columns={columns} _data={data} />);
    const cells = screen.getAllByRole('cell');
    expect(cells[1]).toHaveTextContent('Valor 1');
    expect(cells[2]).toHaveTextContent('Valor 2');
  });

  test('renders Status component', () => {
    const statusColumn = {
      Header: 'Status',
      accessor: 'status',  
    };
    const statusData = [{status: 'Pendente'}];
    render(<Table _columns={[statusColumn]} _data={statusData} />);
    
    const statusComponent = screen.getByType(Status);
    expect(statusComponent).toBeInTheDocument();
  });

  test('renders Loading component if loading=true', () => {
    render(<Table _columns={columns} loading={true} />);
    
    const loadingComponent = screen.getByType(Loading);
    expect(loadingComponent).toBeInTheDocument();
  });

  test('renders Error component if error=true', () => {
    render(<Table _columns={columns} error={true} />);
    
    const errorComponent = screen.getByType(Error);
    expect(errorComponent).toBeInTheDocument(); 
  });
  */

  test('snapshot', () => {
    const {asFragment} = render(<Table _columns={columns} _data={data} />);
    expect(asFragment()).toMatchSnapshot();
  });  

});