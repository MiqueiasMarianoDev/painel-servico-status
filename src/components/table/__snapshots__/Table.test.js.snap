// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table snapshot 1`] = `
<DocumentFragment>
  <table
    role="table"
  >
    <thead>
      <tr
        role="row"
      >
        <th
          colspan="1"
          role="columnheader"
        >
          Coluna 1
        </th>
      </tr>
    </thead>
    <tbody
      role="rowgroup"
    >
      <tr
        role="row"
      >
        <td
          role="cell"
        >
          Valor 1
        </td>
      </tr>
      <tr
        role="row"
      >
        <td
          role="cell"
        >
          Valor 2
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
